Plot size

ConstantinGahr opened the pull request:
Hey there,

second times the charm.

I started using Helium today and realized that images are always scaled to the size of the window. I don't like this and there is also a bug report regarding this.

This pull request scales the figure to the minimum of the window size and the figure size. As such the figure is always displayed as a whole but never larger than the original image.

BR,
C

Fixes #108

sschuhmann commented:
Hi, this looks good. Thank you for the PR!

Should we add a setting to switch between the different behaviors?

ConstantinGahr commented:
Thanks. 

You mean something like
```json
"image_size": "original",
"image_size": "resize_to_window",
"image_size": "optimal",
```
where the latter is the implementation of this PR?

sschuhmann commented:
Yes, this actually looks good. In the merge request a scale factor was proposed, but having these three options should fit for the majority of use cases.

ConstantinGahr commented:
I can look into it.

ConstantinGahr commented:
I updated the settings file with the discussed settings:
```json
  // "image_size" controls the size of the image. It can take one three values:
  //     "original":         The image is displayed with its original size.
  //     "resize_to_window": The image is rescaled to match the size of the window.
  //     "optimal":          The image is either resized or has its original size,
  //                         whichever is smaller.
  // The default is "optimal".
  "image_size": "optimal"
```

Additionally, I added the rescaling of images to non-internal visualization.

Lastly, I sorted and cleaned the settings file.

sschuhmann commented:
Thank you for the PR!
